controllers/osartifact_controller.go

```diff
@@ -99,41 +100,69 @@ func (r *OSArtifactReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 }
 
-func (r *OSArtifactReconciler) startBuild(ctx context.Context, artifact *osbuilder.OSArtifact) (ctrl.Result, error) {
-	// generate configmap required for building a custom image
+// CreateConfigMap generates a configmap required for building a custom image
+func (r *OSArtifactReconciler) CreateConfigMap(ctx context.Context, artifact *osbuilder.OSArtifact) error {
 	cm := r.genConfigMap(artifact)
 	if cm.Labels == nil {
 		cm.Labels = map[string]string{}
 	}
 	cm.Labels[artifactLabel] = artifact.Name
 	if err := controllerutil.SetOwnerReference(artifact, cm, r.Scheme()); err != nil {
-		return ctrl.Result{Requeue: true}, err
+		return err
 	}
 	if err := r.Create(ctx, cm); err != nil && !apierrors.IsAlreadyExists(err) {
-		return ctrl.Result{Requeue: true}, err
+		return err
 	}
 
+	return nil
+}
+
+func (r *OSArtifactReconciler) CreatePVC(ctx context.Context, artifact *osbuilder.OSArtifact) (*corev1.PersistentVolumeClaim, error) {
```

Comment:
Liter seems to not be executing for osbuilder since it doesn't catch the missing comment

Reply:
There was a fix in the reusable workflow here: https://github.com/kairos-io/linting-composite-action/blob/f4640f9aed637bf2c2593516d87d1d702cee49b6/.github/workflows/reusable-linting.yaml#L19

which we never tagged so we are not consuming it. I'll do it now.

Reply:
I now enabled the linter but it seems that the comment rule is not enabled by default: https://www.pixelstech.net/article/1650069870-golangci-lint-to-enable-comment-check-for-exported-functions

Following the instructions in the link above I was able to make it complain with this `.golangci.yaml` :

```
# enable revive linter
linters:
  enable:
    - revive

issues:
  include:
    - EXC0012  # EXC0012 revive: Annoying issue about not having a comment. The rare codebase has such comments
    - EXC0014  # EXC0014 revive: Annoying issue about not having a comment. The rare codebase has such comments
```

but this introduces more (irrelevant) linting errors that need fixing. This belongs to another PR. For now, I will just add comments to the new exported methods in this PR.

Reply:
Actually, I un-exported this. Strictly speaking, it doesn't need a comment anymore.